pkg/provider/azure_routes.go

```diff
@@ -179,6 +180,18 @@ func (d *delayedRouteUpdater) updateRoutes(ctx context.Context) {
 				break
 			}
 		}
+		// After removing the matched routes (if any), loop again to remove the outdated routes,
+		// whose name and IP addresses may not match but target the same node.
+		for i, existingRoute := range routes {
```

Comment:
To reduce the potential impact, I separate the two deletion logic by adding a new for loop, instead of integrating them into one loop.